### PR TITLE
feat(DTFS2-6911): Regression QAT prior deployment

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure QAT suites are executed prior to any deployment.

## Resolution :heavy_check_mark:
* Refactored deployment script to accompany all the pre-defined environments.
* Pre-deployment GitHub event webhook for `test.yml` successful completion on selected branches:
  * dev
  * feature
  * staging
  * production
* Removed environment specific deployment GHA workflows.
* Refactored `test.yml`, to allow execution prior to targeted environment deployment.

## Miscellaneous :heavy_plus_sign:
* Updated dependencies.
* Updated documentation.
* Type casted env variable to `string`.

### CODE REVIEW

**Review by avaitonis — DISMISSED** [review]
LGTM

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+          echo WEBAPP=$(az resource list --resource-type 'Microsoft.Web/sites' --query '[?contains(name, `${{ matrix.webapp }}`)].name' -o tsv) >> $GITHUB_ENV
+
+      - name: ACR 🔐
+        uses: azure/docker-login@v1
+        with:
+          login-server: ${{ env.ACR }}
+          username: ${{ env.ACR_USER }}
+          password: ${{ secrets.ACR_PASSWORD }}
+
+      - name: Directory 📁
```
Comment (ChristianMcc) [inline .github/workflows/deployment.yml:120]: At some point should we just update the name of the directory and microservice to portal-ui? :D (not for this PR ofc)

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    env:
+      ENVIRONMENT: ${{ needs.setup.outputs.environment }}
+
+    strategy:
+      max-parallel: 1
+      # Do not cancel in-progress jobs upon failure
+      fail-fast: false
+      # Single dimension matrix
+      matrix:
+        functions: ["acbs-function", "number-generator-function"]
```
Comment (ChristianMcc) [inline .github/workflows/deployment.yml:185]: Irrelevant for this PR but out of curiosity where is ACBS set up right now? Do dev and staging both point to the same version of TFS API?

File: .github/workflows/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+    strategy:
+      max-parallel: 2
+      # Do not cancel in-progress jobs upon failure
+      fail-fast: false
+      # Single dimension matrix
+      matrix:
+        webapp:
+          [
+            "portal-ui",
```
Comment (ChristianMcc) [inline .github/workflows/deployment.yml:77]: nice refactor from the old scripts!

**Review by ChristianMcc — APPROVED** [review]
Looks good to me!
Only thing is are environment variables set up on github actions/secrets for the dev/staging/prod branches in a way that we need some values for running the docker-compose up in the github pipeline and some variables for deploying to dev/staging/production? Don't have visibility of the secrets/environments so not sure if this is an issue or not :)

**Comment by abhi-markan** [comment]
> Looks good to me! Only thing is are environment variables set up on github actions/secrets for the dev/staging/prod branches in a way that we need some values for running the docker-compose up in the github pipeline and some variables for deploying to dev/staging/production? Don't have visibility of the secrets/environments so not sure if this is an issue or not :)

Thanks for the approval @ChristianMcc docker runtime environment variables are defined in their respective environment, so do not expect this to be an issue. Thanks

**Comment by abhi-markan** [comment]
> Looks good to me! Only thing is are environment variables set up on github actions/secrets for the dev/staging/prod branches in a way that we need some values for running the docker-compose up in the github pipeline and some variables for deploying to dev/staging/production? Don't have visibility of the secrets/environments so not sure if this is an issue or not :)

Thanks for the approval @ChristianMcc docker runtime environment variables are defined in their respective environment, so do not expect this to be an issue. Thanks

**Pull request merged**